staging/src/k8s.io/api/extensions/v1beta1/types.go

```diff
@@ -708,6 +752,11 @@ type IngressBackend struct {
 
 	// Specifies the port of the referenced service.
 	ServicePort intstr.IntOrString `json:"servicePort" protobuf:"bytes,2,opt,name=servicePort"`
+
+	// Resource is an ObjectRef to another Kubernetes resource in the namespace
+	// of the Ingress object.
+	// +optional
+	Resource *v1.TypedLocalObjectReference `json:"resource,omitempty" protobuf:"bytes,3,opt,name=resource"`
```

Comment:
adding a new field that is mutually exclusive to serviceName/servicePort will effectively change those two fields from required to optional, right? they are currently marked as required in the schema, and kubectl will fail to apply manifests that do not contain a serviceName/servicePort field. If we want the schema to reflect that those two fields are considered optional now, add an `+optional` comment to those two fields (applies to all external versions)

That means that eventually (once we start allowing data into the resource field, v1beta1 Ingress objects with a resource backend will also serialize with `serviceName: "", servicePort: 0`. Are ingress controllers prepared to handle that?